guardian/elasticsearch-node-rotation
# make use of `RotateWithElasticsearchNodeRotation` tag more intuitive

As hinted at in the description of https://github.com/guardian/elasticsearch-node-rotation/pull/91, in this PR I aim to make the selection of which node to rotate more intuitive (**but** stopped short of eliminating the tag on the instances all together, in part because its actually used in the CFN... https://github.com/guardian/elasticsearch-node-rotation/blob/88f57c470b5775f36d5d979e537469675f5de702/cloudformation.yaml#L135 ... to restrict SSM access, but also it does provide flexibility about which specific instances are eligible for rotation).

## What does this change?
Remove `GetOldestNode` step/lambda since the behaviour was largely duplicated in the `DiscoverAutoScalingGroup` step (now renamed to `GetTargetNode`) which now...

- finds oldest node based on whether the **ASG has the** `RotateWithElasticsearchNodeRotation` tag set to "true" AND the instance has the same tag set to true - unlike the previous behaviour where it chose the ASG based on whether any instance within it had the tag set to true, then found the oldest within that ASG to actually rotate, regardless of whether that instance had the tag set to true (which is where the confusion arose)
- allows the node to be specified explicitly by `targetInstanceId` in the input object 

## How to test
Although already tested with the grid (see https://github.com/guardian/elasticsearch-node-rotation/pull/92#issuecomment-1010940122), to test simply do a partial deploy of https://riffraff.gutools.co.uk/preview/yaml?project=elasticsearch-node-rotation&build=253&stage=INFRA&updateStrategy=MostlyHarmless with your teams steps selected... then run with and without `targetInstanceId` in the input object to test the two behaviours.

## Images
![image](https://user-images.githubusercontent.com/19289579/148962006-bc26945a-376c-471a-b9e0-030777165e44.png)
